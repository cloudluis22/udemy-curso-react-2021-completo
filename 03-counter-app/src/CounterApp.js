import React, { useState } from 'react';
import PropTypes from 'prop-types'

const CounterApp = ({ number }) => {

    const [ counter, setCounter ] = useState(number);
    
    const addNumber = () => setCounter(counter + 1);
    const substractNumber = () => setCounter(counter - 1);
    const resetNumber = () =>  setCounter(number);
    

    return (
        <>
            <h1> Counter App </h1>
            <h2> { counter } </h2>
            
            <button onClick = { addNumber } id="agregar" > +1 </button>
            <button onClick = { substractNumber } id="restar" > -1 </button>
            <button onClick = { resetNumber } id="reiniciar" > Reiniciar </button>
        </>
    );
}

CounterApp.propTypes = {
    number: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    number: 15 
}

export default CounterApp;