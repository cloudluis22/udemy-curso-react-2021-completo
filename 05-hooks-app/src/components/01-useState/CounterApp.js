import React, { useState } from 'react';
import './counter.css';

export const CounterApp = () => {

    // El el useState tendra dos valores iniciales en vez de solo uno, almacenados dentro de un objeto.
    const [state, setState] = useState({
        counter1: 10,
        counter2: 20
    });

    // Desestructurando las variables de el objeto de el useState inicial. (Esto sirve de referencia para mantener los estados.)
    const { counter1, counter2 } = state

    return (
        <>
            <h1>Counter 1 { counter1 }</h1>
            <h1>Counter 2 { counter2 }</h1>
            <hr/>

            <button
                className="btn btn-primary"
                onClick={ () => {
                    setState({
                        ...state,
                        counter1: counter1 + 1
                    });
                }}
            >
                +1 (Counter 1)
            </button>

            <button
                className="btn btn-primary"
                onClick={() => {
                    setState({
                        ...state,
                        counter2: counter2 + 1
                    });
                }}
            >
                +1 (Counter 2)
            </button>
        </>
    )
}
