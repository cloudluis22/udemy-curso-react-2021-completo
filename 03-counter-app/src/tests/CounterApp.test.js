import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas en el componente del Counter App.', () => {

    let wrapper = shallow(<CounterApp />)

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });
    
    test('Debe hacer match el componente con una snapshot. ', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('El componente de CounterApp debe mostrar el valor por defecto de 100.', () => {

        const number = 100;
        const wrapper = shallow(<CounterApp number={number} />)
        
        const h2Info = Number(wrapper.find('h2').text());
    
        expect(h2Info).toBe(number);

    });
    
    test('El botón +1 de CounterApp debe funcionar correctamente.', () => {

        wrapper.find('button').at(0).simulate('click');
        const h2Info = Number(wrapper.find('h2').text());

        expect(h2Info).toBe(16)
        
    });

    test('El botón -1 de CounterApp debe funcionar correctamente.', () => {

        wrapper.find('button').at(1).simulate('click');

        const h2Info = Number(wrapper.find('h2').text());

        expect(h2Info).toBe(14);

    });
    
    test('El botón de reiniciar de el componente debería funcionar.', () => {

        const number = 64;
        const wrapper = shallow(<CounterApp number={number} />)

        wrapper.find('button').at(1).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        wrapper.find('button').at(2).simulate('click');
        const h2Info = Number(wrapper.find('h2').text());

        expect(h2Info).toBe(number);

    });
    
    
})
