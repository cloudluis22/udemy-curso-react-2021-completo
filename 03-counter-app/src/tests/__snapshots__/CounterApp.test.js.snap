// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente del Counter App. Debe hacer match el componente con una snapshot.  1`] = `
<Fragment>
  <h1>
     Counter App 
  </h1>
  <h2>
     
    15
     
  </h2>
  <button
    id="agregar"
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    id="restar"
    onClick={[Function]}
  >
     -1 
  </button>
  <button
    id="reiniciar"
    onClick={[Function]}
  >
     Reiniciar 
  </button>
</Fragment>
`;
